tests/components/life360/test_config_flow.py

```diff
@@ -0,0 +1,334 @@
+    CONF_AUTHORIZATION,
+    CONF_DRIVING_SPEED,
+    CONF_MAX_GPS_ACCURACY,
+    DEFAULT_OPTIONS,
+    DOMAIN,
+    SHOW_DRIVING,
+)
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+
+from tests.common import MockConfigEntry
+
+TEST_USER = "Test@Test.com"
+TEST_PW = "password"
+TEST_PW_3 = "password_3"
+TEST_AUTHORIZATION = "authorization_string"
+TEST_AUTHORIZATION_2 = "authorization_string_2"
+TEST_AUTHORIZATION_3 = "authorization_string_3"
+TEST_MAX_GPS_ACCURACY = "300"
+TEST_DRIVING_SPEED = "18"
+TEST_SHOW_DRIVING = True
+
+USER_INPUT = {CONF_USERNAME: TEST_USER, CONF_PASSWORD: TEST_PW}
+
+TEST_CONFIG_DATA = {
+    CONF_USERNAME: TEST_USER,
+    CONF_PASSWORD: TEST_PW,
+    CONF_AUTHORIZATION: TEST_AUTHORIZATION,
+}
+TEST_CONFIG_DATA_2 = {
+    CONF_USERNAME: TEST_USER,
+    CONF_PASSWORD: TEST_PW,
+    CONF_AUTHORIZATION: TEST_AUTHORIZATION_2,
+}
+TEST_CONFIG_DATA_3 = {
+    CONF_USERNAME: TEST_USER,
+    CONF_PASSWORD: TEST_PW_3,
+    CONF_AUTHORIZATION: TEST_AUTHORIZATION_3,
+}
+
+USER_OPTIONS = {
+    "limit_gps_acc": True,
+    CONF_MAX_GPS_ACCURACY: TEST_MAX_GPS_ACCURACY,
+    "set_drive_speed": True,
+    CONF_DRIVING_SPEED: TEST_DRIVING_SPEED,
+    SHOW_DRIVING: TEST_SHOW_DRIVING,
+}
+TEST_OPTIONS = {
+    CONF_MAX_GPS_ACCURACY: float(TEST_MAX_GPS_ACCURACY),
+    CONF_DRIVING_SPEED: float(TEST_DRIVING_SPEED),
+    SHOW_DRIVING: TEST_SHOW_DRIVING,
+}
+
+
+# ========== Common Fixtures & Functions ===============================================
+
+
+@pytest.fixture(name="life360", autouse=True)
+def life360_fixture():
+    """Mock life360 config entry setup & unload."""
+    with patch(
+        "homeassistant.components.life360.async_setup_entry", return_value=True
+    ), patch("homeassistant.components.life360.async_unload_entry", return_value=True):
+        yield
+
+
+@pytest.fixture
+def life360_api():
+    """Mock Life360 api."""
+    with patch("homeassistant.components.life360.config_flow.Life360") as mock:
+        yield mock.return_value
+
+
+def create_config_entry(hass, state=None):
+    """Create mock config entry."""
+    config_entry = MockConfigEntry(
+        domain=DOMAIN,
+        data=TEST_CONFIG_DATA,
+        version=1,
+        state=state,
+        options=DEFAULT_OPTIONS,
+        unique_id=TEST_USER.lower(),
+    )
+    config_entry.add_to_hass(hass)
+    return config_entry
+
+
+# ========== User Flow Fixtures & Functions ============================================
+
+
+@pytest.fixture
+async def user_flow_init(hass, life360_api):
+    """Initialize user flow."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    await hass.async_block_till_done()
+    return result
+
+
+async def user_flow(hass, life360_api, flow_id, auth_result=None):
+    """Continue user flow."""
+    life360_api.reset_mock(return_value=True, side_effect=True)
+    if auth_result:
+        if isinstance(auth_result, Exception):
```

Comment:
Please use pytest fixtures and parametrization instead of conditional logic in the tests. We don't want any branches in the tests.

Fixtures can be composed and overridden as needed to get different behavior in the tests.

Reply:
That's ok.